app/controllers/public/index.js

```diff
@@ -29,43 +29,41 @@ export default Controller.extend({
         'password'               : (Math.random() * 10).toString(16),
         'wasRegisteredWithOrder' : true
       });
-      newUser.save()
-        .then(() => {
-          let credentials = newUser.getProperties('email', 'password'),
-              authenticator = 'authenticator:jwt';
-          credentials.identification = newUser.email;
-          this.session
-            .authenticate(authenticator, credentials)
-            .then(async() => {
-              const tokenPayload = this.authManager.getTokenPayload();
-              if (tokenPayload) {
-                this.set('session.skipRedirectOnInvalidation', true);
-                this.authManager.persistCurrentUser(
-                  await this.store.findRecord('user', tokenPayload.identity)
-                );
-                this.set('isLoginModalOpen', false);
-                this.send('placeOrder');
-              }
-            })
-            .catch(reason => {
-              console.warn(reason);
-            })
-            .finally(() => {
-              this.set('session.skipRedirectOnInvalidation', false);
-            });
-        })
-        .catch(error => {
-          if (error.errors[0]) {
-            if (error.errors[0].status === 409) {
-              this.set('userExists', true);
-            } else {
-              this.notify.error(this.l10n.t(error.errors[0].detail));
-            }
+      try {
+        await newUser.save();
+
+        let credentials = newUser.getProperties('email', 'password'),
+            authenticator = 'authenticator:jwt';
+        credentials.identification = newUser.email;
+
+        try {
+          await this.session.authenticate(authenticator, credentials);
+          const tokenPayload = this.authManager.getTokenPayload();
+          if (tokenPayload) {
+            this.set('session.skipRedirectOnInvalidation', true);
+            this.authManager.persistCurrentUser(
+              await this.store.findRecord('user', tokenPayload.identity)
+            );
+            this.set('isLoginModalOpen', false);
+            this.send('placeOrder');
           }
-        })
-        .finally(() => {
-          this.set('isLoading', false);
-        });
+        } catch (reason) {
+          console.warn(reason);
+        } finally {
+          this.set('session.skipRedirectOnInvalidation', false);
+        }
+      } catch (error) {
+        if (error.errors[0]) {
+          if (error.errors[0].status === 409) {
+            this.set('userExists', true);
+          } else {
+            this.notify.error(this.l10n.t(error.errors[0].detail));
+          }
+        }
+      } finally {
+        this.set('isLoading', false);
+      }
```

Comment:
This is not the solution

Reply:
What should be the approach then?

Reply:
This doesn't even fix the issue I mentioned. So the approach should be whatever which doesn't at least break the existing app

Reply:
Wrapping await calls inside try-catch which, if fail were crashing fastboot app server ?

Reply:
No, that did not fix the issue, this is called ignoring the issue, not fixing it. Besides, it is still crashing

Reply:
@iamareebjamal I am not able to reproduce the crashing on my local, I think I am doing something wrong: 
- To enable fastboot I need to set the env flag to false and use `http://localhost:4200/?fastboot=true` ?

Reply:
https://github.com/fossasia/open-event-frontend/pull/3786#issuecomment-574534529

Reply:
Tried it :
- Throws error due to missing package `fastboot-app-server` and `safe-eval`. Installed them .
- Worker starts and throws error : 
```js
 /Users/kush/open-event-frontend/dist/assets/open-event-frontend-fastboot.js:1



SyntaxError: Invalid or unexpected token
    at new Script (vm.js:86:7)
    at VMSandbox.eval (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/vm-sandbox.js:13:22)
    at /Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/ember-app.js:197:15
    at Array.forEach (<anonymous>)
    at EmberApp.loadAppFiles (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/ember-app.js:194:18)
    at EmberApp.retrieveSandboxedApp (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/ember-app.js:235:10)
    at new EmberApp (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/ember-app.js:61:21)
    at FastBoot._buildEmberApp (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/index.js:114:17)
    at new FastBoot (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/node_modules/fastboot/src/index.js:52:10)
    at Worker.buildMiddleware (/Users/kush/open-event-frontend/node_modules/fastboot-app-server/src/worker.js:74:21)
```

Reply:
Open  /Users/kush/open-event-frontend/dist/assets/open-event-frontend-fastboot.js and see what's wrong

Reply:
That file only has hash numbers. How will I debug it 😕 

<img src="https://user-images.githubusercontent.com/44091822/72469385-4ca38c00-3805-11ea-8c07-5382712c406d.png" width="258">

Reply:
It should be JS

Reply:
It is not !

Reply:
@iamareebjamal Should I follow the approach you used in your WIP PR or should I revert the extra changes here and move ahead with wip PR ?

Reply:
First you need to use fastboot server to reproduce the error. Show your .env file please

Reply:
```sh
GOOGLE_API_KEY="Sample Key"
API_HOST=http://127.0.0.1:5000
FASTBOOT_DISABLED=false
```

Reply:
Use yarn start to run the normal server and see the contents of fastboot server JS file. If it is still binary, then delete node_modules folder and reinstall

Reply:
> Use yarn start to run the normal server and see the contents of fastboot server JS file. If it is still binary, then delete node_modules folder and reinstall


Same binary file. Perhaps it may work if I copied your js file into that binary.

Reply:
```js
define("open-event-frontend/initializers/ajax",["exports"],function(e){"use strict"
Object.defineProperty(e,"__esModule",{value:!0}),e.default=void 0
e.default={name:"ajax-service",initialize:function(){}}}),define("open-event-frontend/initializers/error-handler",["exports"],function(e){"use strict"
Object.defineProperty(e,"__esModule",{value:!0}),e.default=void 0
var t={name:"error-handler",initialize:function(e){Ember.onerror||(Ember.onerror=function(e){var t="There was an error running your app in fastboot. More info about the error: \n ".concat(e.stack||e)
Ember.Logger.error(t)})}}
e.default=t}),define("open-event-frontend/instance-initializers/setup-fetch",["exports","fetch"],function(e,t){"use strict"
Object.defineProperty(e,"__esModule",{value:!0}),e.default=void 0
var r={name:"fetch",initialize:function(e){var r=e.lookup("service:fastboot").get("request"),n="undefined:"===r.protocol?"http:":r.protocol;(0,t.setupFastboot)(n,r.get("host"))}}
e.default=r})

```

Reply:
@iamareebjamal It worked! I will test now 😄

Reply:
@iamareebjamal It worked and I tested it a several times. Didn't got any transition error. Everything worked smoothely !

Reply:
Go to fossasia summit 2020, click on login in nav bar. Login with credentials. It'll return you back to the CFS page. And throw an error

Reply:
> Go to fossasia summit 2020, click on login in nav bar. Login with credentials. It'll return you back to the CFS page. And throw an error

Why will go to eventyay ? This change is on my local ?

Reply:
Nobody asked you to go to eventyay. Change API_HOST to https://api.eventyay.com and retest

Reply:
@iamareebjamal Working on fastboot, Also checked on `fossasia summit 2020`. There was an issue in loading currentUser which was resulting into null query. Took reference from your WIP PR. It is working well now

Reply:
This should be reverted

Reply:
The try catch block ?

Reply:
@iamareebjamal done!